test/should-compile/GHC/TypeLits/Test.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE DataKinds, RankNTypes, TypeFamilies, TypeOperators #-}
+{-# LANGUAGE DataKinds, RankNTypes, TypeFamilies, TypeOperators, AllowAmbiguousTypes #-}
```

Comment:
Why is this needed?

Reply:
Because the `proxy s` below should actually be `proxy s1` and `proxy s2`.

I now notice that the examples are inconsistent: some use `proxy` in order to make the type variables unambiguous, while others use `AllowAmbiguousTypes`.

Reply:
fixed in 6501dee5e59a9741346b12b89a66bc3b33caebce.